plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/AppEngineWebMarkerResolutionGenerator.java

```diff
@@ -16,22 +16,41 @@
 
 package com.google.cloud.tools.eclipse.appengine.validation;
 
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
 import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.runtime.CoreException;
 import org.eclipse.ui.IMarkerResolution;
 import org.eclipse.ui.IMarkerResolutionGenerator;
 
 /**
  * Returns possible resolutions for given resource marker. Quick Fix resolutions
  * are available in the Problems view.
  */
-public class ApplicationMarkerResolutionGenerator implements IMarkerResolutionGenerator {
+public class AppEngineWebMarkerResolutionGenerator implements IMarkerResolutionGenerator {
 
+  private static final Logger logger =
+      Logger.getLogger(AppEngineWebMarkerResolutionGenerator.class.getName());
+  
   @Override
   public IMarkerResolution[] getResolutions(IMarker marker) {
     IMarkerResolution[] markerResolutions = new IMarkerResolution[1];
-    IMarkerResolution fix = new ApplicationQuickFix();
-    markerResolutions[0] = fix;
-    return markerResolutions;
+    try {
+      if ("com.google.cloud.tools.eclipse.appengine.validation.applicationMarker"
+          .equals(marker.getType())) {
+        IMarkerResolution fix = new ApplicationQuickFix();
+        markerResolutions[0] = fix;
+      } else if ("com.google.cloud.tools.eclipse.appengine.validation.versionMarker"
+          .equals(marker.getType())) {
+        IMarkerResolution fix = new VersionQuickFix();
+        markerResolutions[0] = fix;
+      }
+      return markerResolutions;
```

Comment:
You can move the return statement after the try block since it's used here and in the catch clause.